addons/account/wizard/account_payment_register.py

```diff
@@ -1092,10 +1096,14 @@ def _create_payments(self):
 
         payments = self._init_payments(to_process, edit_mode=edit_mode)
         self._post_payments(to_process, edit_mode=edit_mode)
-        self._reconcile_payments(to_process, edit_mode=edit_mode)
+
+        if not self.is_register_payment_on_draft:
+            self._reconcile_payments(to_process, edit_mode=edit_mode)
         return payments
 
     def action_create_payments(self):
+        if self.is_register_payment_on_draft:
+            self.payment_difference_handling = 'open'
```

Comment:
Would it be wrong to just filter the draft lines in `filtered_domain` in `_reconcile_payments` instead?
```diff
diff --git a/addons/account/wizard/account_payment_register.py b/addons/account/wizard/account_payment_register.py
index abfb6e25f8b7..ce752c41c9ce 100644
--- a/addons/account/wizard/account_payment_register.py
+++ b/addons/account/wizard/account_payment_register.py
@@ -146,7 +146,6 @@ class AccountPaymentRegister(models.TransientModel):
         compute='_compute_show_require_partner_bank') # used to know whether the field `partner_bank_id` should be required
     country_code = fields.Char(related='company_id.account_fiscal_country_id.code', readonly=True)
     duplicate_move_ids = fields.Many2many(comodel_name='account.move', compute='_compute_duplicate_moves')
-    is_register_payment_on_draft = fields.Boolean(compute='_compute_is_register_payment_on_draft')

     # == trust check ==
     untrusted_bank_ids = fields.Many2many('res.partner.bank', compute='_compute_trust_values')
@@ -688,11 +687,6 @@ class AccountPaymentRegister(models.TransientModel):
             else:
                 wizard.duplicate_move_ids = self.env['account.move']

-    @api.depends('line_ids')
-    def _compute_is_register_payment_on_draft(self):
-        for wizard in self:
-            wizard.is_register_payment_on_draft = any(l.parent_state == 'draft' for l in wizard.line_ids)
-
     def _fetch_duplicate_reference(self, matching_states=('draft', 'posted')):
         """ Retrieve move ids for possible duplicates of payments. Duplicates moves:
         - Have the same partner_id, amount and date as the payment
@@ -1032,7 +1026,11 @@ class AccountPaymentRegister(models.TransientModel):
             for account in payment_lines.account_id:
                 (payment_lines + lines)\
                     .with_context(**extra_context)\
-                    .filtered_domain([('account_id', '=', account.id), ('reconciled', '=', False)])\
+                    .filtered_domain([
+                        ('account_id', '=', account.id),
+                        ('reconciled', '=', False),
+                        ('parent_state', '=', 'posted'),
+                    ])\
                     .reconcile()

     def _create_payments(self):
@@ -1097,13 +1095,10 @@ class AccountPaymentRegister(models.TransientModel):
         payments = self._init_payments(to_process, edit_mode=edit_mode)
         self._post_payments(to_process, edit_mode=edit_mode)

-        if not self.is_register_payment_on_draft:
-            self._reconcile_payments(to_process, edit_mode=edit_mode)
+        self._reconcile_payments(to_process, edit_mode=edit_mode)
         return payments

     def action_create_payments(self):
-        if self.is_register_payment_on_draft:
-            self.payment_difference_handling = 'open'
         payments = self._create_payments()

         if self._context.get('dont_redirect_to_payments'):
diff --git a/addons/account/wizard/account_payment_register_views.xml b/addons/account/wizard/account_payment_register_views.xml
index d69231a8cafd..33bd2edab4ea 100644
--- a/addons/account/wizard/account_payment_register_views.xml
+++ b/addons/account/wizard/account_payment_register_views.xml
@@ -81,7 +81,7 @@
                             <label for="payment_difference"/>
                             <div>
                                 <field name="payment_difference"/>
-                                <field name="payment_difference_handling" widget="radio" nolabel="1" invisible="is_register_payment_on_draft"/>
+                                <field name="payment_difference_handling" widget="radio" nolabel="1"/>
                                 <div invisible="hide_writeoff_section or payment_difference_handling == 'open'">
                                     <label for="writeoff_account_id" string="Post Difference In" class="oe_edit_only"/>
                101701 Liquidity Transfer                     <field name="writeoff_account_id"
```

I quickly tested and everything seems to work, even the write-off feature :shrug:

Reply:
@william-andre 
Yup good idea for the filtering, but we'd still need the computed field `is_register_payment_on_draft` to hide the `payment_difference_handling` field and always set it to `'open'`, nope ? Not sure why you're removing this part in your diff

Reply:
No, that's what I meant by everything seems to work fine with the write-off feature

Reply:
The point was that we didn't want to allow the write-off feature for this case, but I can double-check with TSB tho

Reply:
@william-andre 
TSB confirmed that we don't want to use the write-off feature so we should keep the payment diff open (and so hide the option). So I've kept the computed field for this part but used your suggestion for the filter domain